File: backend/SmartService/Domain/Entities/Reparation.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace SmartService.Domain.Entities
{
    public class Reparation
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey("User")]
        public int UserId { get; set; }

        [ForeignKey("Mechanic")]
        public int MechanicId { get; set; }

        [ForeignKey("Car")]
        public int CarId { get; set; }

        [Required]
        [MaxLength(255)]
        public string Type { get; set; }

        [Required]
        [MaxLength(255)]
        public string Description { get; set; }

        [Required]
        public DateTime Date { get; set; }

        [Required]
        [MaxLength(100)]
        public string Status { get; set; }

        public Car Car { get; set; }
        public User User { get; set; }
        public User Mechanic { get; set; }
        public Bill Bill { get; set; }
    }
}

File: frontend/src/app/models/entities/Reparation.ts
import { Bill } from './Bill';
import { Car } from './Car';
import { User } from './User';

export class Reparation {
    id: number = 0;
    userId: number = 0;
    mechanicId: number = 0;
    carId: number = 0;
    type: string = '';
    description: string = '';
    date: Date = new Date();
    status: string = '';
}
